sky/resources.py

```diff
@@ -1117,12 +1117,92 @@ def get_required_cloud_features(
         return features
 
     @classmethod
-    def from_yaml_config(cls, config: Optional[Dict[str, str]]) -> 'Resources':
-        if config is None:
-            return Resources()
-
+    def from_yaml_config(
+        cls, config: Optional[Dict[str, Any]]
+    ) -> Union[Set['Resources'], List['Resources']]:
         common_utils.validate_schema(config, schemas.get_resources_schema(),
```

Comment:
I'm slightly confused about why we could still use the old schema check. Should we add the `any_of` and `ordered` fields in the resources schema?

Reply:
We already have the `any_of` and `ordered` field included in the schema in #1389